portal-impl/src/com/liferay/portal/servlet/filters/etag/ETagUtil.java

```diff
@@ -29,14 +29,18 @@ public static boolean processETag(
 			return false;
 		}
 
-		int hashCode = _hashCode(
-			byteBuffer.array(), byteBuffer.position(), byteBuffer.limit());
+		String eTag = httpServletResponse.getHeader(HttpHeaders.ETAG);
 
-		String eTag = StringBundler.concat(
-			StringPool.QUOTE, StringUtil.toHexString(hashCode),
-			StringPool.QUOTE);
+		if (eTag == null) {
+			int hashCode = _hashCode(
+				byteBuffer.array(), byteBuffer.position(), byteBuffer.limit());
 
-		httpServletResponse.setHeader(HttpHeaders.ETAG, eTag);
+			eTag = StringBundler.concat(
+				StringPool.QUOTE, StringUtil.toHexString(hashCode),
+				StringPool.QUOTE);
+
+			httpServletResponse.setHeader(HttpHeaders.ETAG, eTag);
+		}
```

Comment:
With this change any other filter/servlet could use `setHeader(HttpHeaders.ETAG)'  instead of `addHeader(HttpHeaders.ETAG)` I see this PR changes that too :+1: , so we are avoiding any consequence derived from running LanguageFilter before EtagFilter

Also, with this change we assume that the Etag is not being calculated everytime we call the method. I don't know if this can be an issue as, theoretically, Etag filter kicks in once, however I wanted to note it just in case.